src/Microsoft.ML.Data/Scorers/ClusteringScorer.cs

```diff
@@ -101,7 +101,9 @@ protected override Delegate GetPredictedLabelGetter(Row output, out Delegate sco
 
             long cachedPosition = -1;
             VBuffer<Float> score = default(VBuffer<Float>);
-            int scoreLength = Bindings.PredColType.GetKeyCount();
+            ulong keyCount = Bindings.PredColType is KeyType key ? key.Count : 0;
+            Contracts.Assert(keyCount <= int.MaxValue);
```

Comment:
Can this be replaced with AssertRangeReturnKeyCount here as well?

Reply:
Yes! Fixed.

---
In reply to: [249208483](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249208483) [](ancestors = 249208483)